index.js

```diff
@@ -299,15 +299,21 @@ module.exports = {
       files = files.filter((filename) => filename.includes('__addonLocation__'));
     }
 
-    if (options.typescript) {
-      return files;
-    } else {
+    if (!options.typescript) {
       let ignoredFiles = ['__addonLocation__/tsconfig.json'];
 
-      return files.filter(
+      files = files.filter(
         (filename) => !filename.match(/.*\.ts$/) && !ignoredFiles.includes(filename)
       );
     }
+
+    if (this.isExistingMonorepo) {
+      // The .gitignore is used for ignoring files that are moved to the addon from the root folder at build time
+      // But this setup does not apply for an existing monorepo (all root files are ignored), so we don't need the .gitignore
+      files = files.filter((filename) => filename !== '__addonLocation__/gitignore');
```

Comment:
in an existing monorepo we do want to keep the addon's gitignore tho.

each package which can generated ignored files benefits from a local gitignore

Maybe I'm missing something?

Reply:
The situation changed a bit with merging #136. And actually I am not sure the changed to the addon's .gitignore [here](https://github.com/embroider-build/addon-blueprint/pull/136/files#diff-808204a18982944927b2d4d3fb95b22316ac767ead202c8219c68d0d9c3ce179) were what we really want. I probably missed the fact that this was the addon's local .gitignore, and not the root one...

Previously, the local .gitignore only had this:
```
/README.md
/LICENSE.md
```

This was necessary to ignore the files that we copy as part of the build [here](https://github.com/embroider-build/addon-blueprint/blob/main/files/__addonLocation__/rollup.config.mjs#L59-L63).

However, in an in-monorepo situation, this is different. We don't copy these files anymore as part of the build, but move them into the addon folder when the blueprint runs. This is then their intended location. But then we must not gitignore them anymore. And as stated above, when those two entries are the only ones in the .gitignore, then taht file does not make any sense anymore. That's why it is not created in this PR.

Back to the changes in #136: I think they should be reverted, that's what I'm doing at #141. Let's discuss than topic over there, in case you disagree...

Reply:
Given that #141 has been merged, the change here makes sense again I think. The addon's gitignore again contains only the Readme and License files, which in an existing monorepo must not be ignored as explained above, so we can delete the addon's .gitignore. The "Introduce workspace isolation" PR we talked about will need to adjust that eventually...